### PR TITLE
Solution review from Octane lead

### PR DESCRIPTION
**Pull request by mike-north**

### CODE REVIEW

File: app/routes/teams/team.js [hunk 1]
```diff
@@ -0,0 +1,11 @@
+import Route from '@ember/routing/route';
+import { ALL_TEAMS } from '../teams';
+
+export default class TeamsTeamRoute extends Route {
+  model({ teamId }) {
+    const matches = ALL_TEAMS.filter(
```
Comment (rwjblue) [inline app/routes/teams/team.js]: Seems like you really want `.find` here, no?

File: app/services/auth.js [hunk 2]
```diff
@@ -11,15 +13,25 @@ export default class AuthService extends Service {
...
   @service router;
 
+  @tracked currentUser = null;
+
   /**
    *
    * @param {string} userId
    */
-  loginWithUserId(userId) {
+  async loginWithUserId(userId) {
```
Comment (rwjblue) [inline app/services/auth.js]: Doesn't use `await` in this method (no need for async), seems odd to add it when it isn't needed?

File: app/routes/login.js [hunk 3]
```diff
@@ -1,4 +1,17 @@
 import Route from '@ember/routing/route';
+import { inject as service } from '@ember/service';
+import AuthService from 'shlack/services/auth';
```
[the remark this answers is not in the document]
Comment (mike-north) [reply app/routes/login.js]: Correct, this is just imported for the comment. This is the way I usually do this, in situations where I'm certain it won't increase bundle size or trigger lazy behavior of some sort.

I'm open to alternatives, but haven't evaluated pros/cons yet.

File: app/routes/teams/team.js (hunk 1 above)
Comment (mike-north) [reply app/routes/teams/team.js]: is this polyfilled for ie11 by default?

File: app/services/auth.js (hunk 2 above)
Comment (mike-north) [reply app/services/auth.js]: Good catch. There should be a `await this.loadCurrentUser()` in there

File: app/templates/components/team-sidebar.hbs [hunk 4]
```diff
@@ -46,11 +46,9 @@
   </nav>
 
   <footer class="mx-4 mb-2 text-white">
-    <LinkTo @route='login' {{! destination route }}
-            @tagName="button" {{! use <button> instead of <a> }}
-            class="text-white rounded bg-grey-dark hover:bg-red-darker p-2 team-sidebar__logout-button" {{! HTML classes}}
+    <button {{on "click" this.auth.logout}} class="text-white rounded bg-grey-dark hover:bg-red-darker p-2 team-sidebar__logout-button"
```
[the remark this answers is not in the document]
Comment (mike-north) [reply app/templates/components/team-sidebar.hbs]: I think the JS module is created in 
Step 10 - https://github.com/mike-north/ember-octane-workshop/commit/63af3489958899216ce0d30607421bd156fbfc34#diff-4115de6334f96ca58c6ffb037073289b
and this application of the `{{on` modifier is introduced in 
Step 12 - https://github.com/mike-north/ember-octane-workshop/commit/4dd715de37f7dd1d4b2f2e354cb154bc26710fcc#diff-c8e3fdaeb0c5a86900d5ee286d2c9819

File: app/components/channel-container.js [hunk 5]
```diff
@@ -0,0 +1,89 @@
...
+
+  /**
+   * @type {AuthService}
+   */
+  @service auth;
+
+  @action
+  async loadMessages() {
+    const {
+      channel: { id: channelId, teamId },
```
[the remark this answers is not in the document]
Comment (mike-north) [reply app/components/channel-container.js]: In my own code I like nested destructuring (probably left over preference from working w/ pattern matching). However, I completely agree with you for a workshop project like this. Easier-to-understand is strictly better.

File: app/routes/login.js (hunk 3 above)
[the remark this answers is not in the document]
Comment (rwjblue) [reply app/routes/login.js]: Nope, seems totally fine I just hadn't seen the practice before so I was curious.

File: app/templates/components/team-sidebar.hbs (hunk 4 above)
[the remark this answers is not in the document]
Comment (rwjblue) [reply app/templates/components/team-sidebar.hbs]: Yeah, I think I got confused by the GH UI when reviewing commit by commit. Basically, I wasn't reviewing in the right order (sorry about that).

File: app/routes/teams/team.js (hunk 1 above)
Comment (rwjblue) [reply app/routes/teams/team.js]: No, it wouldn't be. If we are targeting IE11 we should stick with `.filter()[0]`.

Comment (mike-north) [reply app/routes/teams/team.js]: I think I'll leave it as-is, so that the workshop is operable on all of the browsers ember supports by default

**Pull request closed without merging**